src/Phoenix/PhoenixClient.ts

```diff
@@ -14,15 +14,27 @@ import { userContext } from "../UserContext";
 import { getAuthorizationHeader } from "../Utils/AuthorizationUtils";
 
 export class PhoenixClient {
-  public async containerConnectionInfo(
-    provisionData: IProvisionData
+  private containerHealthHandler: NodeJS.Timeout;
+
+  public async allocateContainer(provisionData: IProvisionData): Promise<IResponse<IPhoenixConnectionInfoResult>> {
+    return this.getContainerInfo(provisionData, "allocate");
+  }
+
+  public async resetContainer(provisionData: IProvisionData): Promise<IResponse<IPhoenixConnectionInfoResult>> {
+    return this.getContainerInfo(provisionData, "reset");
+  }
+
+  private async getContainerInfo(
```

Comment:
The name of the function doesn't seem to represent what it does anymore. Consider renaming.